.github/workflows/goreleaser.yaml

```diff
@@ -52,7 +52,7 @@ jobs:
         run: echo "version_flags=$(./scripts/version-ldflags)" >> "$GITHUB_OUTPUT"
       - name: Run GoReleaser
         id: run-goreleaser
-        uses: goreleaser/goreleaser-action@5742e2a039330cbb23ebf35f046f814d4c6ff811 # v5.1.0
+        uses: goreleaser/goreleaser-action@286f3b13b1b49da4ac219696163fb8c1c93e1200 # v6.0.0
```

Comment:
We will have to fix some deprecation notices, as mentioned in the migration guide
https://goreleaser.com/blog/goreleaser-v2/#upgrading

Reply:
Made the required changes in [these 4 commits](https://github.com/ossf/scorecard/pull/4158/files/850c236fbe9ce848e1274b653fe9cc10389fbbb1..5d670aeb2a8926062861d96194d8e0eb0b3cd0ab) if I could get a second set of eyes @ossf/scorecard-maintainers.

After the changes, both `goreleaser check` and `make build-releaser` work.